source/ref_trainings.rst

```diff
@@ -0,0 +1,269 @@
+
+********************************************
+MLFlow
+********************************************
+
+`MLflow <https://mlflow.org/docs/latest/index.html>`_ is library-agnostic. You can use it with any machine learning library, and in any programming language, since all functions are accessible through a REST API and CLI.
+
+Installation
+------------
+
+The most straightforward way to install the MLFlow trainer on an Odahu Cluster is to deploy the `odahu-flow-mlflow` helm chart.
+The helm chart registers the trainer in the API Odahu and deploys an `a MLflow Tracking Server <https://www.mlflow.org/docs/latest/tracking.html#mlflow-tracking-servers>`_.
+By default, the deployed MLflow Tracking Server is available at `https://cluster-url/mlflow` address.
+
+.. code-block:: bash
+
+    # Add the odahu-flow helm repository
+    helm repo add odahu-flow 'https://raw.githubusercontent.com/odahu/odahu-helm/master/'
+    helm repo update
+    # Fill in the values for the chart or leave the default values
+    helm inspect values odahu-flow/odahu-flow-mlflow --version 1.0.0 > values.yaml
+    vim values.yaml
+    # Deploy the helm chart
+    helm install odahu-flow/odahu-flow-mlflow --name odahu-flow-mlflow --namespace odahu-flow --debug -f values.yaml --atomic --wait --timeout 120
+
+.. warning::
+
+    Odahu-flow must be deployed before the mlflow trainer.
+
+MLProject file
+--------------
+
+Let's look at how the MLProject file is related to Model Training API.
+
+.. code-block:: text
+
+    name: My Project
+
+    entry_points:
+      main:
+        parameters:
+          data_file: path
+          regularization: {type: float, default: 0.1}
+        command: "python train.py -r {regularization} {data_file}"
+      test:
+        parameters:
+          data_file: path
+        command: "python validate.py {data_file}"
+
+Model Training API only can contain one entry point.
+You have to add all hyperparameters, which do not have a default value, to a Model Training.
+Next, you can find the Model Trainings for the MLProject file.
+
+.. code-block:: yaml
+
+    spec:
+      entrypoint: main
+      hyperParameters:
+        data_file: test/123.txt
+        regularization: 0.2
+
+.. code-block:: yaml
+
+    spec:
+      entrypoint: main
+      hyperParameters:
+        data_file: test/456.txt
+
+.. code-block:: yaml
+
+    spec:
+      entrypoint: test
+      hyperParameters:
+        data_file: test/123.txt
+
+MLFlow protocol
+---------------
+
+Odahu-flow requires that a model be logged through `mlflow API <https://www.mlflow.org/docs/latest/python_api/mlflow.pyfunc.html#mlflow.pyfunc.log_model>`_.
+
+.. code-block:: python
+    :caption: Example of sklearn model logging:
+
+    mlflow.sklearn.log_model(lr, "model")
+
+Optionally, you can provide input and output samples for Odahu-flow.
+It allows determining input and output types for Odahu-flow packagers.
```

Comment:
```suggestion
It allows to determine input and output types for Odahu-flow packagers.
```

Reply:
Grammarly recommends the gerund form in the this case.